src/components/Thread.js

```diff
@@ -51,6 +51,16 @@ const Thread = withChannelContext(
          *  **Available from [channel context](https://getstream.github.io/stream-chat-react-native/#channelcontext)**
          * */
         Message: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
+        /**
+         * **Customized MessageInput component to used within Thread instead of default MessageInput
+         * **Available from [MessageInput](https://getstream.github.io/stream-chat-react-native/#messageinput)**
+         * */
+        MessageInput: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
```

Comment:
@mahboubii it should be `PropTypes.oneOfType([PropTypes.node, PropTypes.elementType])`